repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -0,0 +1,362 @@
+import org.openrepose.nodeservice.request.RequestHeaderService
+import org.openrepose.nodeservice.response.ResponseHeaderService
+import org.openrepose.powerfilter.ReposeRoutingServlet._
+import org.springframework.beans.factory.annotation.Value
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+@Named
+class ReposeRoutingServlet @Inject()(@Value(ReposeSpringProperties.NODE.CLUSTER_ID) clusterId: String,
+                                     @Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                                     configurationService: ConfigurationService,
+                                     requestHeaderService: RequestHeaderService,
+                                     responseHeaderService: ResponseHeaderService,
+                                     reportingService: ReportingService,
+                                     optMetricsService: Optional[MetricsService]
+                                    ) extends HttpServlet with UpdateListener[SystemModel] with StrictLogging {
+
+  private val localCluster = new AtomicReference[Option[ReposeCluster]](None)
+  private val localNode = new AtomicReference[Option[Node]](None)
+  private val metricsService = new AtomicReference[Option[MetricsService]](Option(optMetricsService.orElse(null)))
+  private val defaultDestination = new AtomicReference[Option[Destination]](None)
+  private val destinations = new AtomicReference[Map[String, Destination]](Map.empty[String, Destination])
+  private val domains = new AtomicReference[Option[Clusters]](None) // TODO: Remove this when routing to a cluster is removed.
+  private var initialized = false
+
+  override def init(): Unit = {
+    logger.info("{}:{} -- Reticulating Splines - Initializing Repose Routing Servlet", clusterId, nodeId)
+    configurationService.subscribeTo(
+      SystemModelConfigurationFilename,
+      getClass.getResource("/META-INF/schema/system-model/system-model.xsd"),
+      this,
+      classOf[SystemModel]
+    )
+  }
+
+  override def destroy(): Unit = {
+    configurationService.unsubscribeFrom(SystemModelConfigurationFilename, this)
+    logger.info("{}:{} -- Obfuscated Quigley Matrix - Destroying Repose Routing Servlet", clusterId, nodeId)
+  }
+
+  override def configurationUpdated(configurationObject: SystemModel): Unit = {
+    //Set the current system model, and just update the nodes.
+    val interrogator = new SystemModelInterrogator(clusterId, nodeId)
+    localCluster.set(Option(interrogator.getLocalCluster(configurationObject).orElse(null)))
+    localNode.set(Option(interrogator.getLocalNode(configurationObject).orElse(null)))
+    defaultDestination.set(Option(interrogator.getDefaultDestination(configurationObject).orElse(null)))
+    destinations.set(localCluster.get().map(_.getDestinations) match {
+      case Some(dests) => getDestinations(dests.getEndpoint.asScala.toList) ++ getDestinations(dests.getTarget.asScala.toList)
```

Comment:
Why call `getDestinations` twice rather than just joining the lists and calling it once? Also, why convert to a `List` explicitly?

I might recommend:
```
getDestinations(dests.getEndpoint.asScala++ dests.getTarget.asScala)
```

Reply:
Fair enough.
Updated.

Reply:
Ok. That made it a couple of lines.